### PR TITLE
feat: modern ping syntax

### PR DESCRIPTION
**Pull request by ghost**
Fixes #

#### Proposed Changes

- Modern Ping Command

### CODE REVIEW

File: src/commands/general/ping.ts [hunk 1]
```diff
@@ -4,7 +4,12 @@ export const ping: CommandInterface = {
   name: "ping",
   description: "Pings the bot.",
   command: async (message) => {
-    await message.reply("Pong!");
-    return;
+    message.reply('Pinging...').then((reply) => {
```
Comment (eddiejaoude) [inline src/commands/general/ping.ts:7]: we are using `async` instead of `then`, better to be consistent

**Comment by eddiejaoude** [comment]
Thank you for the improvement 👍 , could you add a screenshot to the resulting changes, thanks 🤓

**Comment by krishdevdb** [comment]
> Thank you for the improvement 👍 , could you add a screenshot to the resulting changes, thanks 🤓

@eddiejaoude there can't be a screenshot but I can give you a example message:

```md
**User:** !stargate ping
**Stargate:** Pinging...
```

Within 1 second (varies on bot speed)

```md
**Stargate(edits message):** Pong The message round-trip took ${(reply.editedTimestamp || reply.createdTimestamp) - (message.editedTimestamp || message.createdTimestamp)}ms.  The heartbeat ping is ${Math.round(message.client.ws.ping)}ms.` : ''}
```

**Comment by krishdevdb** [comment]
This is a builtin commando command, also the modern ping command syntax, since we are not using commando we need to implement it from scratch

**Comment by eddiejaoude** [comment]
Thank you for making those changes.

CI failed with lint errors https://github.com/eddiejaoude/stargate/pull/132/checks?check_run_id=3054428736

```
/home/runner/work/stargate/stargate/src/commands/general/ping.ts
Error:    9:21   error  Replace `message.channel.type·!==·'dm'·?·`${message.author},`·:·''` with `⏎······message.channel.type·!==·"dm"·?·`${message.author},`·:·""⏎····`                                                                                                        prettier/prettier
Error:    9:46   error  Strings must use doublequote                                                                                                                                                                                                                            quotes
Error:    9:76   error  Strings must use doublequote                                                                                                                                                                                                                            quotes
Error:   10:54   error  Replace `(reply.editedTimestamp·||·reply.createdTimestamp)·-·(message.editedTimestamp·||·message.createdTimestamp)` with `⏎······(reply.editedTimestamp·||·reply.createdTimestamp)·-⏎······(message.editedTimestamp·||·message.createdTimestamp)⏎····`  prettier/prettier
Error:   11:19   error  Replace `message.client.ws.ping·?·`The·heartbeat·ping·is·${Math.round(message.client.ws.ping)}ms.`·:·''` with `⏎······message.client.ws.ping⏎········?·`The·heartbeat·ping·is·${Math.round(message.client.ws.ping)}ms.`⏎········:·""⏎····`              prettier/prettier
Error:   11:111  error  Strings must use doublequote                                                                                                                                                                                                                            quotes

✖ 6 problems (6 errors, 0 warnings)
```

**Comment by eddiejaoude** [comment]
>  there can't be a screenshot

not sure what you mean, can you not take a screenshot of when you tested it in Discord?

**Comment by krishdevdb** [comment]
> > there can't be a screenshot
> 
> not sure what you mean, can you not take a screenshot of when you tested it in Discord?

Yep, but I mean that somethings can't have screenshots as they are dynamic?

**Comment by krishdevdb** [comment]
> Thank you for making those changes.
> 
> CI failed with lint errors https://github.com/eddiejaoude/stargate/pull/132/checks?check_run_id=3054428736
> 
> ```
> /home/runner/work/stargate/stargate/src/commands/general/ping.ts
> Error:    9:21   error  Replace `message.channel.type·!==·'dm'·?·`${message.author},`·:·''` with `⏎······message.channel.type·!==·"dm"·?·`${message.author},`·:·""⏎····`                                                                                                        prettier/prettier
> Error:    9:46   error  Strings must use doublequote                                                                                                                                                                                                                            quotes
> Error:    9:76   error  Strings must use doublequote                                                                                                                                                                                                                            quotes
> Error:   10:54   error  Replace `(reply.editedTimestamp·||·reply.createdTimestamp)·-·(message.editedTimestamp·||·message.createdTimestamp)` with `⏎······(reply.editedTimestamp·||·reply.createdTimestamp)·-⏎······(message.editedTimestamp·||·message.createdTimestamp)⏎····`  prettier/prettier
> Error:   11:19   error  Replace `message.client.ws.ping·?·`The·heartbeat·ping·is·${Math.round(message.client.ws.ping)}ms.`·:·''` with `⏎······message.client.ws.ping⏎········?·`The·heartbeat·ping·is·${Math.round(message.client.ws.ping)}ms.`⏎········:·""⏎····`              prettier/prettier
> Error:   11:111  error  Strings must use doublequote                                                                                                                                                                                                                            quotes
> 
> ✖ 6 problems (6 errors, 0 warnings)
> ```

How do I replace template strings with double quotes might be a bug in ci?

**Comment by eddiejaoude** [comment]
> How do I replace template strings with double quotes might be a bug in ci?

It is suggesting to replace single quotes with doubles. It also shows examples of the suggestion.

> Replace \`(reply.editedTimestamp·||·reply.createdTimestamp)·-·(message.editedTimestamp·||·message.createdTimestamp)` with `⏎······(reply.editedTimestamp·||·reply.createdTimestamp)·-⏎······(message.editedTimestamp·||·message.createdTimestamp)⏎····\`

This is the message broken out....

Replace `(reply.editedTimestamp·||·reply.createdTimestamp)·-·(message.editedTimestamp·||·message.createdTimestamp)` 

with `⏎······(reply.editedTimestamp·||·reply.createdTimestamp)·-⏎······(message.editedTimestamp·||·message.createdTimestamp)⏎····`

**Comment by eddiejaoude** [comment]
I think it needs to be something like this

```ts
export const ping: CommandInterface = {
  name: "ping",
  description: "Pings the bot.",
  command: async (message) => {
    const reply = await message.reply("Pinging...");

    const ping = `${
      message.channel.type !== "dm" ? `${message.author},` : ""
    } Pong!`;
    const roundTrip = `The message round-trip took ${
      (reply.editedTimestamp || reply.createdTimestamp) -
      (message.editedTimestamp || message.createdTimestamp)
    }ms.`;
    const ws = `${
      message.client.ws.ping
        ? `The heartbeat ping is ${Math.round(message.client.ws.ping)}ms.`
        : ""
    }`;

    reply.edit(`${ping} ${roundTrip} ${ws}`);

    return;
  },
};
```

**Comment by krishdevdb** [comment]
> I think it needs to be something like this
> 
> ```ts
> export const ping: CommandInterface = {
>   name: "ping",
>   description: "Pings the bot.",
>   command: async (message) => {
>     const reply = await message.reply("Pinging...");
> 
>     const ping = `${
>       message.channel.type !== "dm" ? `${message.author},` : ""
>     } Pong!`;
>     const roundTrip = `The message round-trip took ${
>       (reply.editedTimestamp || reply.createdTimestamp) -
>       (message.editedTimestamp || message.createdTimestamp)
>     }ms.`;
>     const ws = `${
>       message.client.ws.ping
>         ? `The heartbeat ping is ${Math.round(message.client.ws.ping)}ms.`
>         : ""
>     }`;
> 
>     reply.edit(`${ping} ${roundTrip} ${ws}`);
> 
>     return;
>   },
> };
> ```

Thank you Eddie but I fixed all the issues using Eslint directly, no I will test it on discord since there are a lot of changes since the first change

**Comment by krishdevdb** [comment]
Screenshot
discordapp com_channels_847115152885022720_859448346564362310

**Comment by eddiejaoude** [comment]
Great, thank you for the example screenshot 👍

**Review by eddiejaoude — COMMENTED** [review]
Looks good 👍

**Comment by ghost** [comment]
@eddiejaoude I've tested the feature.  It is not there yet? Is auto-deployment working?

**Comment by eddiejaoude** [comment]
Looks like the deployment failed https://github.com/eddiejaoude/stargate/actions

I will run it again, but might need a further look into why when I get time if it fails again

**Comment by ghost** [comment]
> Looks like the deployment failed https://github.com/eddiejaoude/stargate/actions
> 
> I will run it again, but might need a further look into why when I get time if it fails again

@eddiejaoude as per the action it was a timeout, we can rerun it or you can increase the timeout cap

```bash
Error: error: deployment "stargate-deployment" exceeded its progress deadline
```

**Comment by eddiejaoude** [comment]
This is the error from the logs, I do not think increasing the timeout would help, there is an error when starting the container.

```
Error from server (BadRequest): container "stargate" in pod "stargate-deployment-7cdd4bfb65-p8twl" is waiting to start: CreateContainerConfigError
```

I will have to run the container locally and check why it is not running

**Comment by eddiejaoude** [comment]
I fixed it, it was my fault, I deleted an environment variable on the Kubernetes cluster 🤦‍♂️ 

https://github.com/eddiejaoude/stargate/actions/runs/1026926364

I think there is a bug though...

Screenshot 2021-07-14 at 09 24 55

**Comment by ghost** [comment]
Seems like the dev copy and prod copy use different discord.js versions. Fixed in #134 But consider making the dev and prod dependency versions the same

**Comment by eddiejaoude** [comment]
They should be the same, because they will use the package.lock` file

**Comment by ghost** [comment]
> They should be the same, because they will use the package.lock` file

When it runs `npm install` if you have any ~ in the dependency versions the versions can bump

**Pull request merged**